File: src/main/java/nl/chimpgamer/networkmanager/extensions/discordbot/tasks/CreateTokenTask.java
package nl.chimpgamer.networkmanager.extensions.discordbot.tasks;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import net.dv8tion.jda.api.entities.Message;
import net.dv8tion.jda.api.entities.MessageChannel;
import nl.chimpgamer.networkmanager.common.utils.Methods;
import nl.chimpgamer.networkmanager.extensions.discordbot.DiscordBot;
import nl.chimpgamer.networkmanager.extensions.discordbot.utils.DCMessage;
import nl.chimpgamer.networkmanager.extensions.discordbot.utils.JsonEmbedBuilder;
import nl.chimpgamer.networkmanager.extensions.discordbot.utils.Utils;

@RequiredArgsConstructor
@Getter
public class CreateTokenTask implements Runnable {
    private final DiscordBot discordBot;
    private final MessageChannel channel;
    private final String discordID;

    @Override
    public void run() {
        final String token = Utils.generateToken();
        final String msgStr = DCMessage.REGISTRATION_TOKEN.getMessage()
                .replace("%newline%", "\n")
                .replace("%token%", token);

        Message message;

        if (Methods.isJsonValid(msgStr)) {
            JsonEmbedBuilder jsonEmbedBuilder = JsonEmbedBuilder.fromJson(msgStr);
            message = Utils.sendMessageComplete(this.getChannel(), jsonEmbedBuilder.build());
        } else {
            message = Utils.sendMessageComplete(this.getChannel(), msgStr);
        }

        this.getDiscordBot().getDiscordUserManager().insertToken(token, this.getDiscordID(), message);
    }
}

File: src/main/kotlin/nl/chimpgamer/networkmanager/extensions/discordbot/tasks/CreateTokenTask.kt
package nl.chimpgamer.networkmanager.extensions.discordbot.tasks

import net.dv8tion.jda.api.events.interaction.command.GenericCommandInteractionEvent
import nl.chimpgamer.networkmanager.extensions.discordbot.DiscordBot
import nl.chimpgamer.networkmanager.extensions.discordbot.configurations.DCMessage
import nl.chimpgamer.networkmanager.extensions.discordbot.modals.JsonMessageEmbed
import nl.chimpgamer.networkmanager.extensions.discordbot.utils.Utils
import nl.chimpgamer.networkmanager.extensions.discordbot.utils.Utils.generateToken

class CreateTokenTask(
    private val discordBot: DiscordBot,
    private val commandInteractionEvent: GenericCommandInteractionEvent
) : Runnable {

    override fun run() {
        val token = generateToken()
        val msgStr = discordBot.messages.getString(DCMessage.REGISTRATION_TOKEN_RESPONSE)
            .replace("%newline%", "\n")
            .replace("%token%", token)

        if (Utils.isJsonValid(msgStr)) {
            val jsonMessageEmbed = JsonMessageEmbed.fromJson(msgStr)
            commandInteractionEvent.replyEmbeds(jsonMessageEmbed.toMessageEmbed()).setEphemeral(true)
                .queue({ success ->
                    discordBot.discordUserManager.insertToken(token, commandInteractionEvent.user.id,
                    success)
                }) { failure ->
                    discordBot.logger.warning("Failed to send REGISTRATION_TOKEN_RESPONSE!")
                    discordBot.logger.warning(failure.localizedMessage)
                }
        } else {
            commandInteractionEvent.reply(msgStr).setEphemeral(true)
                .queue({ success ->
                    discordBot.discordUserManager.insertToken(token, commandInteractionEvent.user.id,
                        success)
                }) { failure ->
                    discordBot.logger.warning("Failed to send REGISTRATION_TOKEN_RESPONSE!")
                    discordBot.logger.warning(failure.localizedMessage)
                }
        }
    }
}